libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -397,6 +417,40 @@ struct intrinsic_invoker_impl<Ret, std::tuple<array_ptr<T>, size_t, Inputs...>,
    }
 };
 
+/**
+ * Specialization for transcribing  a null_terminated_ptr type in the native method signature
+ * This type transcribes 1 wasm parameters: a char pointer which is validated to contain
+ * a null value before the end of the allocated memory.
+ *
+ * @tparam Ret - the return type of the native method
+ * @tparam Inputs - the remaining native parameters to transcribe
+ * @tparam Translated - the list of transcribed wasm parameters
+ */
+template<typename Ret, typename... Inputs, typename ...Translated>
+struct intrinsic_invoker_impl<Ret, std::tuple<null_terminated_ptr, Inputs...>, std::tuple<Translated...>> {
+   using next_step = intrinsic_invoker_impl<Ret, std::tuple<Inputs...>, std::tuple<Translated..., I32>>;
+   using then_type = Ret(*)(wasm_interface &, null_terminated_ptr, Inputs..., Translated...);
+
+   template<then_type Then>
+   static Ret translate_one(wasm_interface &wasm, Inputs... rest, Translated... translated, I32 ptr) {
+      auto mem = getDefaultMemory(intrinsics_accessor::get_context(wasm).code.instance);
+      if(!mem)
+         Runtime::causeException(Exception::Cause::accessViolation);
+      char *base          = (char*)(getMemoryBaseAddress(mem) + ptr);
```

Comment:
Are we checking for ptr < 0 anywhere?

Reply:
And similarly for the array_ptr<...> commits that happened earlier, we need a check for ptr < 0?

Reply:
Brian go ahead and make these updates with your changes. Since you need these for your work and they are approved, will merge now.

Reply:
I agree, this needs to be fixed (in a new PR). However, I think I'd rather modify the invoker to treat "pointers" as unsigned.